### ISSUE TITLE
Corrupt SIF file should be closed before the exception is raised

### ISSUE THREAD

**Issue by willberriss** [fujiisoup/sif_parser#41]
https://github.com/fujiisoup/sif_parser/blob/a714fc05f05fbabe2b4d21bc099e2c4b5fffc5ab/sif_parser/sif_open.py#L75

The function np_open() should close the file handle if "will_close = True" AND a corrupt file is encountered. It must close the file before raising the ValueError at line 75, otherwise there will be a file handle leak.

My scenario:
When I encounter a corrupt SIF file I catch the ValueError and I would like to move the corrupt SIF file out of the way. But I cannot move it because in order to do for it to be movable, sif_parser.np_open() needs to close the file handle it owns for the file.

**Comment by fujiisoup** [comment]
Hi, @willberriss 
Thank you for raising an issue.

I am trying to fix it in #42, but I am not able to make a test for this issue.
What I tried is 
```python
    try:
        data, info = sif_parser.np_open(filename)
    except (ValueError, SyntaxError):
        with open(filename, 'a') as f:
            assert f.writable()
```
(with filename is just a random non-SIF file), but it passes the test.

Could you suggest a test which fails in the current form?

**Comment by fujiisoup** [comment]
Could share an example file? Github allows to attach a .zip file. This may ease the fix a lot.

**Comment by willberriss** [comment]
The machine I am on doesn't have internet access so it's not so easy to share.  I can tell you the fix I did locally and I can test your fix too if that helps:

sif_open.py

Line no.:
74. if not ignore_corrupt:
75.     if will_close:   # THE FIX
76.         f.close        # THE FIX
77.     raise ValueError(
78.         'The file might be corrupt...

**Comment by willberriss** [comment]
Hi. I've managed to get a sample corrupt SIF file. I attach it here. Hope it helps. 
[c0rrupt.sif.zip](https://github.com/user-attachments/files/17706037/c0rrupt.sif.zip)

**Comment by fujiisoup** [comment]
Thank you.

I am currently working to make the test *fail*. It is working for windows, but not linux. 
[https://github.com/fujiisoup/sif_parser/issues/41#issuecomment-2468506006](This) does not seem working for this case...

**Comment by fujiisoup** [comment]
Fixed in #42
